### PR TITLE
Added reference to Attribute parameter types

### LINKED ISSUE #22699: No list of valid parameter types
https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0655 links to this page saying "See [Attributes](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/attributes/) for a discussion of valid parameter types for an attribute." But nowhere on this page is there a list of valid parameter types. I would have expected the https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/attributes/#attribute-parameters section to contain a list of what parameter types are valid (e.g., the fact that `decimal` is not a valid parameter type is [apparently surprising to people](https://stackoverflow.com/questions/3192833/why-decimal-is-not-a-valid-attribute-parameter-type)), but that list is missing. Another place where a list of valid parameter types would be useful is in [https://docs.microsoft.com/en-us/dotnet/standard/attributes/writing-custom-attributes#declaring-properties](https://docs.microsoft.com/en-us/dotnet/standard/attributes/writing-custom-attributes#declaring-properties).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 23c570f1-2f76-be1b-410a-ed4bd67e33cc
* Version Independent ID: dd329ad3-ca29-61e5-8753-d57792a88516
* Content: [Attributes (C#)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/attributes/)
* Content Source: [docs/csharp/programming-guide/concepts/attributes/index.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/concepts/attributes/index.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by sunilvijendra**
## Summary

Added reference to "Attribute Parameter types section" (https://github.com/dotnet/csharplang/blob/main/spec/attributes.md#attribute-parameter-types) to the following pages:

Attributes: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/attributes/#attribute-parameters
Writing Custom Attributes:  https://docs.microsoft.com/en-us/dotnet/standard/attributes/writing-custom-attributes#see-also

Fixes #22699

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=23831) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=23831) before we can accept your contribution.<br/><br/>:x: sunilvijendra [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=23831)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=23831) it.</sub>

**Comment by sunilvijendra** [comment]
@BillWagner, @tdykstra ,
This is my first PR in this project! Please check the relative link to C# language specification. Added it based on references in other pages!

File: docs/csharp/programming-guide/concepts/attributes/index.md [hunk 1]
```diff
@@ -50,6 +50,8 @@ Many attributes have parameters, which can be positional, unnamed, or named. Any
 
 The first parameter, the DLL name, is positional and always comes first; the others are named. In this case, both named parameters default to false, so they can be omitted. Positional parameters correspond to the parameters of the attribute constructor. Named or optional parameters correspond to either properties or fields of the attribute. Refer to the individual attribute's documentation for information on default parameter values.
 
+For more information on allowed parameter types refer to [Attributes](../../../../../_csharplang/spec/attributes.md#attribute-parameter-types) page in [C# Language Specification](../../../../../_csharplang/spec/introduction.md)
```
Comment (Youssef1313) [inline docs/csharp/programming-guide/concepts/attributes/index.md:53]:
```suggestion
For more information on allowed parameter types, see the [Attributes](~/_csharplang/spec/attributes.md#attribute-parameter-types) section of the [C# language specification](~/_csharplang/spec/introduction.md)
```

File: docs/standard/attributes/writing-custom-attributes.md [hunk 2]
```diff
@@ -154,3 +154,4 @@ To design your own custom attributes, you do not need to master many new concept
 - <xref:System.Attribute?displayProperty=nameWithType>
 - <xref:System.AttributeUsageAttribute?displayProperty=nameWithType>
 - [Attributes](index.md)
+- [Attribute parameter types](../../../_csharplang/spec/attributes.md#attribute-parameter-types)
```
Comment (Youssef1313) [inline docs/standard/attributes/writing-custom-attributes.md:157]:
```suggestion
- [Attribute parameter types](~/_csharplang/spec/attributes.md#attribute-parameter-types)
```

**Review by BillWagner — APPROVED** [review]
Thank you for adding these @sunilvijendra 

We appreciate it!  

Also, thanks to @Youssef1313 for helping with the preferred link syntax.

This looks great, and I'll :shipit: now.

You should see the changes on our live site in the next day or two on our regular publishing cycle. Thanks again!  🎆

**Pull request merged**